### PR TITLE
WT-13104 Ignore permission issues faced by background compaction

### PR DESCRIPTION
**Pull request by etienneptl**

### CODE REVIEW

File: src/conn/conn_compact.c [hunk 1]
```diff
@@ -258,18 +258,21 @@ __background_compact_should_skip(WT_SESSION_IMPL *session, const char *uri, int6
...
         return (0);
-    }
-
-    WT_RET(ret);
-
-    if (file_size <= WT_MEGABYTE) {
-        WT_STAT_CONN_INCR(session, background_compact_skipped);
-        *skipp = true;
-        return (0);
+    } else if (ret == EACCES) {
```
Comment (quchenhao) [inline src/conn/conn_compact.c:265]: Why we are not skipping this file in this case?

Comment (etienneptl) [reply src/conn/conn_compact.c:265]: I think you are right that we should skip - I am just not sure if we get permission issues for the wrong reasons when using that specific function?
What do you think @Sean04 ?

**Review by quchenhao — APPROVED** [review]
lgtm

File: src/conn/conn_compact.c (hunk 1 above)
Comment (Sean04) [reply src/conn/conn_compact.c]: Yes, I think we should skip. Background compact's purpose is to compact eligible files. Files we don't have access to are not eligible and should be skipped.

**Review by Sean04 — APPROVED** [review]
lgtm

**Pull request merged**